code/modules/projectiles/guns/energy/modular.dm

```diff
@@ -0,0 +1,282 @@
+/obj/item/weapon/gun/energy/laser/prototype
+	name = "laser prototype"
+	desc = "A custom prototype laser weapon."
+	icon_state = "energykill"
+	item_state = "energykill"
+	fire_sound = 'sound/weapons/Laser.ogg'
+	slot_flags = SLOT_BELT|SLOT_BACK
+	w_class = 3
+	force = 10
+	origin_tech = list(TECH_COMBAT = 3, TECH_MAGNET = 2)
+	matter = list(DEFAULT_WALL_MATERIAL = 2000)
+	can_turret = 0
+	zoomdevicename = null
+	max_shots = 0
+	burst_delay = 0
+	var/origin_chassis
+	var/gun_type
+	var/list/gun_mods = list()
+	var/obj/item/laser_components/capacitor/capacitor
+	var/obj/item/laser_components/focusing_lens/focusing_lens
+	var/obj/item/laser_components/modulator/modulator
+	var/chargetime
+	var/is_charging
+	var/criticality = 1 //multiplier for the negative effects of capacitor failures. Not just limited to critical failures.
+	var/named = 0
+	var/described = 0
+
+/obj/item/weapon/gun/energy/laser/prototype/examine(mob/user)
+	..(user)
+	if(gun_mods.len)
+		for(var/obj/item/laser_components/modifier/modifier in gun_mods)
+			user << "You can see a \the [modifier] attached."
+	if(capacitor)
+		user << "You can see a \the [capacitor] attached."
+	if(focusing_lens)
+		user << "You can see a \the [focusing_lens] attached."
+
+/obj/item/weapon/gun/energy/laser/prototype/attackby(var/obj/item/weapon/D as obj, var/mob/user as mob)
```

Comment:
Don't use `as X` syntax outside of verbs/procs as verbs. It doesn't do anything.